applications/acoustic_conservation_equations/normal_incidenting_wave/application.h

```diff
@@ -0,0 +1,239 @@
+
+        for(const auto & face : tria.active_face_iterators())
+          if(face->at_boundary())
+            face->set_boundary_id(1);
+
+        tria.refine_global(global_refinements);
+      };
+
+    GridUtilities::create_triangulation<dim>(
+      grid, this->mpi_comm, this->param.grid, lambda_create_triangulation, {});
+
+    GridUtilities::create_mapping(mapping,
+                                  this->param.grid.element_type,
+                                  this->param.mapping_degree);
+  }
+
+  void
+  set_boundary_descriptor() final
+  {
+    this->boundary_descriptor->admittance_bc.insert(
+      std::make_pair(1, std::make_shared<dealii::Functions::ConstantFunction<dim>>(admittance, 1)));
+  }
+
+  void
+  set_field_functions() final
+  {
+    this->field_functions->initial_solution_pressure =
+      std::make_shared<InitialConditionPressure<dim>>();
+
+    this->field_functions->initial_solution_velocity =
+      std::make_shared<dealii::Functions::ZeroFunction<dim>>(dim);
+
+    this->field_functions->right_hand_side =
+      std::make_shared<dealii::Functions::ZeroFunction<dim>>(1);
+  }
+
+  std::shared_ptr<PostProcessorBase<dim, Number>>
+  create_postprocessor() final
+  {
+    PostProcessorData<dim> pp_data;
+
+    // write output for visualization of results
+    pp_data.output_data.time_control_data.is_active  = this->output_parameters.write;
+    pp_data.output_data.time_control_data.start_time = this->param.start_time;
+    pp_data.output_data.time_control_data.trigger_interval =
+      (this->param.end_time - this->param.start_time) / 20.0;
+    pp_data.output_data.directory          = this->output_parameters.directory + "vtu/";
+    pp_data.output_data.filename           = this->output_parameters.filename;
+    pp_data.output_data.write_pressure     = true;
+    pp_data.output_data.write_velocity     = true;
+    pp_data.output_data.write_higher_order = true;
+    pp_data.output_data.degree             = this->param.degree_u;
+
+    // To be able to compute an analytical solution for testing purposes we have to make
+    // multiple assumptions. The application is intended to be used with different
+    // admittances which contradicts the assumptions. Therefore, we compute the analytical
+    // solution only if we run a test and fetch this information from the input file.
+    if(is_test)
+    {
+      // To be able to compute an error and thus test the behavior of the admittance BC for
+      // Y=1 (first-oder ABC). We also have to ensure the that the wave left the domain
+      // when computing the error.
+      AssertThrow(std::abs(admittance - 1.0) < 1.0e-12,
+                  dealii::ExcMessage("The analytical solution only holds for Y=1."));
+      AssertThrow(this->param.end_time > 1.5 * radius / speed_of_sound,
+                  dealii::ExcMessage("The end time has to be after the wave left the domain."));
+
+      // calculation of pressure error
```

Comment:
Velocity does not make sense here?

Reply:
[f6db7a7](https://github.com/exadg/exadg/pull/629/commits/f6db7a71b09ace5010d6443011cf38242c4702bb)